coalib/misc/Caching.py

```diff
@@ -0,0 +1,142 @@
+import os
+import calendar
+import time
+import appdirs
+import pickle
```

Comment:
@hypothesist what happens if coala is interrupted while writing to the pickled file? Will this break things (and therefore permanently break coala until a human intervenes?)

^@sils1297